python_modules/dagster/dagster/_core/definitions/asset_check_result.py

```diff
@@ -137,13 +137,19 @@ def resolve_target_check_key(
     def to_asset_check_evaluation(
         self, step_context: "StepExecutionContext"
     ) -> AssetCheckEvaluation:
-        check_names_by_asset_key = (
-            step_context.job_def.asset_layer.check_names_by_asset_key_by_node_handle.get(
-                step_context.node_handle.root
-            )
+        assets_def_for_check = check.not_none(
```

Comment:
just as an idea (feel free to ignore), it could make sense to push this logic down into the asset layer, just because this is a fairly in-depth operation to be doing on the AssetCheckResult object.